server/server.go

```diff
@@ -101,15 +101,15 @@ type Server struct {
 	watcher           *watch.Watcher
 	decisionIDFactory func() string
 	diagnostics       Buffer
-	revision          string
+	revisions         map[string]string
 	logger            func(context.Context, *Info) error
 	errLimit          int
 	pprofEnabled      bool
 	runtime           *ast.Term
 	httpListeners     []httpListener
-	bundleStatus      bundle.Status
+	bundleStatuses    map[string]*bundle.Status
 	bundleStatusMtx   *sync.RWMutex
-	hasBundle         bool
+	bp                *bundle.Plugin
```

Comment:
Nit: this doesn't have to be addressed in this PR but I wonder if it would be simpler to just lookup the bundle plugin on the manager as needed instead of caching it on the server. Ditto for the status.

Reply:
I updated it for the bundle plugin but kept the status. The reasoning is that we don't have a nice way to just grab the current status from the plugin, we have to listen for updates from it asynchronously. To add an API to just get the current status we would need to do some extra locking and/or copying of them which I figure we could avoid by stashing the latest update in the background.